cmd/compile/compile.go

```diff
@@ -0,0 +1,164 @@
+package compile
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/flyteorg/flytectl/cmd/register"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/flyteorg/flytepropeller/pkg/compiler"
+	"github.com/flyteorg/flytepropeller/pkg/compiler/common"
+	"github.com/golang/protobuf/proto"
+	"github.com/spf13/cobra"
+)
+
+// Utility function for compiling a list of Tasks
+func compileTasks(tasks []*core.TaskTemplate) ([]*core.CompiledTask, error) {
+	res := make([]*core.CompiledTask, 0, len(tasks))
+	for _, task := range tasks {
+		compiledTask, err := compiler.CompileTask(task)
+		if err != nil {
+			return nil, err
+		}
+		res = append(res, compiledTask)
+	}
+	return res, nil
+}
+
+/*
+ Utility to compile a packaged workflow locally.
+ compilation is done locally so no flyte cluster is required.
+*/
+func compileFromPackage(packagePath string) error {
+	args := []string{packagePath}
+	fileList, tmpDir, err := register.GetSerializeOutputFiles(context.Background(), args, true)
+	defer os.RemoveAll(tmpDir)
+	if err != nil {
+		fmt.Println("Error found while extracting package..")
+		return err
+	}
+	fmt.Println("Successfully extracted package...")
+	fmt.Println("Processing Protobuff files...")
+	workflows := make(map[string]admin.WorkflowSpec)
+	plans := make(map[string]admin.LaunchPlan)
+	tasks := []admin.TaskSpec{}
+
+	for _, pbFilePath := range fileList {
+		// deserializing packaged protocolbuffers
+		// based on filename as defined in flytekit
+		// https://github.com/flyteorg/flytekit/blob/master/flytekit/tools/serialize_helpers.py#L102
+		// filename suffix tell us how to deserialize file
+		// _1.pb : are Tasks protobufs
+		// _2.pb : are workflows protobufs
+		// _3.pb are Launchplans protobufs
+
```

Comment:
Instead of having this logic, can we use the similar logic as used during registration  https://github.com/flyteorg/flytectl/blob/5a17906593f9c6f4bcdb57268037aab768eb0724/cmd/register/register_util.go#L95